cmd/kyma/alpha/deploy/dnsconfigurer.go

```diff
@@ -0,0 +1,77 @@
+package deploy
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+)
+
+// LocalKymaDevDomain is the local Kyma domain name (used as default domain if Kyma is installed locally)
+const LocalKymaDevDomain = "local.kyma.dev"
+
+// DNSConfigurer modifies the Kubernetes DNS system to resolve *.local.kyma.dev names to locally running
+// services. This is only required if Kyma runs locally (e.g. in a local K3s cluster) and uses the local Kyma dev domain.
+//
+// Per default, the DNS zone local.kyma.dev is managed by a public DNS system (which resolves names always to 127.0.0.1)
+// to simplify the local setup of Kyma. But some sub-domains (e.g dex.local.kyma.dev) have to point to locally running
+// Kubernetes services and have to return their internal IP address. This requires a patch of the Kubernetes DNS resolver.
+type DNSConfigurer struct {
+	kubeClient kubernetes.Interface
+}
+
+// NewDNSConfigurer creates a new DNS configurer instance
+func NewDNSConfigurer(kubeClient kubernetes.Interface) *DNSConfigurer {
+	return &DNSConfigurer{
+		kubeClient: kubeClient,
+	}
+}
+
+// ConfigureCoreDNS patches the Kubernetes CoreDNS to support the local kyma domain.
+// It adds the rewrite plugin of CoreDNS to delegate requests to *.local.kyma.dev to the Istio Ingress Gateway.
+func (kld *DNSConfigurer) ConfigureCoreDNS() error {
```

Comment:
Do we really need a type with a single method? Or can this be just a package function?
If there are plans to make it configurable and with different utilities for DNS stuff I could see a type being useful, but at the moment it just adds boilerplate without benefit.

Reply:
Good point, I usually prefer types to be a bit more flexible how to handle data. But agree, in this case it unnecessary code.

Reply:
FIXED

Reply:
It is always possible to turn this function into a method should we need it, but for now it does not hold any data or state.

Reply:
agree